--- pyconfig.h.orig	2020-10-24 22:13:50.745086908 +0100
+++ pyconfig.h	2020-10-24 23:04:46.647877114 +0100
@@ -979,7 +979,7 @@
 #define HAVE_SYS_FILE_H 1
 
 /* Define to 1 if you have the <sys/ioctl.h> header file. */
-#define HAVE_SYS_IOCTL_H 1
+//#define HAVE_SYS_IOCTL_H 1
 
 /* Define to 1 if you have the <sys/kern_control.h> header file. */
 /* #undef HAVE_SYS_KERN_CONTROL_H */
--- Modules/Setup.orig	2020-10-24 23:04:17.224003619 +0100
+++ Modules/Setup	2020-10-24 23:04:24.079974258 +0100
@@ -358,7 +358,7 @@
 # Andrew Kuchling's zlib module.
 # This require zlib 1.1.3 (or later).
 # See http://www.gzip.org/zlib/
-#zlib zlibmodule.c -I$(prefix)/include -L$(exec_prefix)/lib -lz
+zlib zlibmodule.c -IModules/zlib -LModules/zlib -lz
 
 # Interface to the Expat XML parser
 #
--- setup.py
+++ setup.py
@@ -16,7 +16,7 @@ from distutils.command.install_lib import install_lib
 from distutils.command.build_scripts import build_scripts
 from distutils.spawn import find_executable
 
-cross_compiling = "_PYTHON_HOST_PLATFORM" in os.environ
+cross_compiling = True
 
 # Add special CFLAGS reserved for building the interpreter and the stdlib
 # modules (Issue #21121).
