diff --git a/pyconfig.h.orig b/pyconfig.h
--- a/pyconfig.h
+++ b/pyconfig.h
@@ -269,7 +269,7 @@
 
 /* Define if we can use gcc inline assembler to get and set x87 control word
    */
-#define HAVE_GCC_ASM_FOR_X87 1
+#undef HAVE_GCC_ASM_FOR_X87
 
 /* Define if you have the getaddrinfo function. */
 #define HAVE_GETADDRINFO 1
diff --git a/setup.py b/setup.py
--- a/setup.py
+++ b/setup.py
@@ -18,6 +18,7 @@ from distutils.command.install_lib import install_lib
 from distutils.spawn import find_executable
 
 cross_compiling = "_PYTHON_HOST_PLATFORM" in os.environ
+cross_compiling = True
 
 def get_platform():
     # cross build
@@ -442,7 +443,9 @@ class PyBuildExt(build_ext):
             add_dir_to_list(self.compiler.include_dirs, '/usr/local/include')
         if cross_compiling:
             self.add_gcc_paths()
-        self.add_multiarch_paths()
+        #self.add_multiarch_paths()
+        add_dir_to_list(self.compiler.library_dirs, './Modules/zlib')
+        add_dir_to_list(self.compiler.include_dirs, './Modules/zlib')
 
         # Add paths specified in the environment variables LDFLAGS and
         # CPPFLAGS for header and library files.
diff --git a/Modules/Setup b/Modules/Setup
--- a/Modules/Setup
+++ b/Modules/Setup
@@ -465,6 +465,7 @@ GLHACK=-Dclear=__GLclear
 # This require zlib 1.1.3 (or later).
 # See http://www.gzip.org/zlib/
 #zlib zlibmodule.c -I$(prefix)/include -L$(exec_prefix)/lib -lz
+zlib zlibmodule.c -I./Modules/zlib -L./Modules/zlib -lz
 
 # Interface to the Expat XML parser
 #
